src/tests/hash_functions/test_hash_functions.f90

```diff
@@ -0,0 +1,313 @@
+module test_hash_functions
+    use testdrive, only : new_unittest, unittest_type, error_type, check, skip_test
+    use stdlib_kinds, only: int8, int32, int64, dp
+    use stdlib_32_bit_hash_codes, only: little_endian &
+        , nmhash32 &
+        , nmhash32x &
+        , water_hash
+    use stdlib_64_bit_hash_codes, only: pengy_hash, spooky_hash
+ 
+    implicit none
+    private
+    public :: collect_hash_functions
+    public :: generate_key_array
+
+    integer, parameter :: size_key_array = 2048
+
+    integer(int32), parameter :: nm_seed = int( z'deadbeef', int32 )
+    integer(int64), parameter :: water_seed = int( z'deadbeef1eadbeef', int64 )
+    integer(int32), parameter :: pengy_seed = int( z'deadbeef', int32 )
+    integer(int64), parameter :: spooky_seed(2) = [ water_seed, water_seed ]
+
+    interface read_array
+        module procedure read_array_int8
+        module procedure read_array_int32
+        module procedure read_array_int64
+        module procedure read_2darray_int64
+    end interface
+
+contains
+
+    !> Collect all exported unit tests
+    subroutine collect_hash_functions(testsuite)
+        !> Collection of tests
+        type(unittest_type), allocatable, intent(out) :: testsuite(:)
+
+        testsuite = [ &
+            new_unittest("dummy", test_dummy) &
+            , new_unittest("little_endian", test_little_endian) &
+            , new_unittest("nmhash32", test_nmhash32) &
+            , new_unittest("nmhash32x", test_nmhash32x) &
+            , new_unittest("water_hash", test_water_hash) &
+            , new_unittest("pengy_hash", test_pengy_hash) &
+            , new_unittest("spooky_hash", test_spooky_hash) &
+            ]
+
+    end subroutine collect_hash_functions
+
+    subroutine test_dummy(error)
+        !> Error handling
+        type(error_type), allocatable, intent(out) :: error
+
+        call check(error, .true., "dummy")
+        if (allocated(error)) return
+
+    end subroutine
```

Comment:
I would prefer that the `end subroutine`, both here and elsewhere include the procedure name. If that is time consuming I can easily add it myself in emacs.

Reply:
Done

Reply:
looks good.